ops/docker-sequencer-test/run-ha-test.sh

```
@@ -0,0 +1,1693 @@
+
+run_p2p_opt_tests() {
+    log_section "Category 7: P2P Broadcast Reactor Optimization Tests"
+
+    # TC-P2P-01: fullnode applies blocks from HA sequencer (end-to-end sync path).
+    # The fullnodes (node-0/1/2/3, sentry-node-0) use broadcast_reactor.go's
+    # applyRoutine. If it works, they will stay within a few blocks of the HA
+    # leader. We give a 10s window and require delta >= 1.
+    log_info "--- TC-P2P-01: fullnode applies blocks via P2P ---"
+    local leader_height_before follower_height_before
+    local leader_height_after follower_height_after
+    leader_height_before=$(get_block_number "$HA_RPC_NODE0")
```

Shellcheck (0.11.0)

[warning] leader_height_before appears unused. Verify use (or export if used externally).

(SC2034)